test/benchmarks.cc

```diff
@@ -135,7 +138,7 @@ TEST_CASE("benchmark task handling", "[benchmark][task]") {
 struct task_manager_benchmark_context {
 	const size_t num_nodes = 1;
 	task_manager tm{1, nullptr};
-	test_utils::mock_buffer_factory mbf{tm};
+	std::unique_ptr<test_utils::mock_buffer_factory> mbf{std::make_unique<test_utils::mock_buffer_factory>(tm)};
```

Comment:
If I'm not mistaken, this can remain a `test_utils::mock_buffer_factory` without the `std::unique_ptr` indirection.

Reply:
These need to be uniform across the different contexts so that the benchmarking code can always use `->` for member access!

Reply:
AFAICS the `mbf`s of all benchmark contexts can be values.